### PR TITLE
[3.4] Add testing of etcd in local image in release workflow

### PR DESCRIPTION
**Pull request by pchan**
Cherry pick of #15505 on release-3.4.

#15505: scripts: Add testing of etcd in local image in release

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -24,3 +25,6 @@ jobs:
         Expire-Date: 0
         EOF
         DRY_RUN=true ./scripts/release.sh --no-upload --no-docker-push --in-place 3.4.99
+    - name: test-image
+      run: |
+        VERSION=3.6.99 ./scripts/test_images.sh
```
Comment (ahrtr) [inline .github/workflows/release.yaml:30]:
```suggestion
        VERSION=3.4.99 ./scripts/test_images.sh
```

**Comment by pchan** [comment]
I see failures on invoking `scripts/build_lib.sh`, will debug and update.

**Comment by ahrtr** [comment]
@pchan could you resolve this PR? I am planning to release 3.4.25 soon. Will raise a ticket to track the release plan soon.

File: scripts/test_images.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    echo "Please use linux machine for release builds."
+    exit 1
+fi
+
+# Pick defaults based on release workflow
+ARCH=$(go env GOARCH)
+REPOSITARY=${REPOSITARY:-"gcr.io/etcd-development/etcd"}
+if [ -n "$VERSION" ]; then
+    # Expected Format: v3.6.99-amd64
+    TAG=v"${VERSION}"-"${ARCH}"
```
Comment (ahrtr) [inline scripts/test_images.sh:43]: There is no suffix "-amd64" for amd64 arch in 3.4. 
```suggestion
    if [ "${ARCH}" != "amd64" ]; then
        TAG=v"${VERSION}"-"${ARCH}"
    else
        TAG=v"${VERSION}"
    fi
```

Comment (pchan) [reply scripts/test_images.sh]: Just pushed a version that removes the suffix, this one is better.

File: scripts/test_images.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    exit 1
+fi
+
+# Pick defaults based on release workflow
+ARCH=$(go env GOARCH)
+REPOSITARY=${REPOSITARY:-"gcr.io/etcd-development/etcd"}
+if [ -n "$VERSION" ]; then
+    if [ "${ARCH}" != "amd64" ]; then
+		# Expected Format: v3.4.99-arm64
+        TAG=v"${VERSION}"-"${ARCH}"
```
Comment (ahrtr) [inline scripts/test_images.sh:44]:
```suggestion
	# Expected Format: v3.4.99-arm64
        TAG=v"${VERSION}"-"${ARCH}"
```

File: scripts/test_images.sh [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    else
+		# Expected Format: v3.4.99
+        TAG=v"${VERSION}"
```
Comment (ahrtr) [inline scripts/test_images.sh:47]:
```suggestion
	# Expected Format: v3.4.99
        TAG=v"${VERSION}"
```

**Comment by pchan** [comment]
Release passes now.

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @pchan

**Pull request merged**